### PR TITLE
✨ [Story Preview] Enable stories to play when in preview mode

### PR DESCRIPTION
**Pull request by coreymasanto**
- Part of #37129 

---

The layout logic of `amp-story` is designed to block until the story has reached the `visible` visibility state. However, the story should also layout and play in the newly introduced `preview` visibility state.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story.js [hunk 1]
```diff
@@ -1041,11 +1042,14 @@ export class AmpStory extends AMP.BaseElement {
...
-    // visible.
+    // previewed or visible.
     const initialPageEl = this.element.querySelector(
       `amp-story-page#${escapeCssSelectorIdent(initialPageId)}`
     );
-    if (!this.getAmpDoc().hasBeenVisible()) {
+    if (
+      !this.getAmpDoc().hasBeenVisible() &&
+      !this.getAmpDoc().hasBeenPreviewed()
+    ) {
```
Comment (coreymasanto) [inline extensions/amp-story/1.0/amp-story.js:1052]: I'm unsure whether this change is necessary

File: extensions/amp-video/0.1/amp-video.js [hunk 2]
```diff
@@ -397,7 +397,7 @@ export class AmpVideo extends AMP.BaseElement {
         this.video_.setAttribute('preload', 'auto');
       }
       pendingOriginPromise = this.getAmpDoc()
-        .whenFirstVisible()
+        .whenFirstPreviewedOrVisible()
```
Comment (jridgewell) [inline extensions/amp-video/0.1/amp-video.js:400]: I believe this is a privacy leak. Shouldn't we only be connecting to an origin domain when we become visible?

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: extensions/amp-video/0.1/amp-video.js (hunk 2 above)
Comment (gmajoulet) [reply extensions/amp-video/0.1/amp-video.js:400]: +1, `amp-video` will correctly play cached sources in `prerender` or `preview` mode, and then if needed play the origin sources when `visible`. I think this is the expected behavior, we shouldn't have to change anything to `amp-video`.

File: extensions/amp-story/1.0/amp-story.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js:1052]: I think it's not needed. To verify, plz check if the promise `return storyLayoutPromise;` actually resolves (i.e. if `storyLayoutPromise.then(() => console.log('hehe'));` logs when loading the page in `preview` vState, you don't need this change.

**Review by jridgewell — APPROVED** [review]

**Review by gmajoulet — APPROVED** [review]

**Pull request merged**